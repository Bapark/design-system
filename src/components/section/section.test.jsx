import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Section from './section.jsx';

describe('Section', () => {
  it('renders a section component', () => {
    const tree = renderer
      .create((
        <Section title="Test Title" />
      )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses a title', () => {
    render(<Section title="Test Title" />);
    expect(screen.getByText('Test Title').tagName).toEqual('H2');
  });

  it('can have a description', () => {
    render(<Section title="Test Title" description="This is a test description." />);
    expect(screen.getByText('This is a test description.').tagName).toEqual('P');
  });

  it('can have child elements', () => {
    render(<Section title="Test Title" ><span>Test Label</span></Section>);
    expect(screen.getByText('Test Label').tagName).toEqual('SPAN');
  });
});
