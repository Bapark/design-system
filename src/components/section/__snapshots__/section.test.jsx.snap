// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders a section component 1`] = `
<section
  className="outer-section"
>
  <h2
    className="subheading"
  >
    Test Title
  </h2>
</section>
`;
